lib/puppet/pops/evaluator/compare_operator.rb

```diff
@@ -149,4 +155,64 @@ def include_Array(a, b, scope)
   def include_Hash(a, b, scope)
     include?(a.keys, b, scope)
   end
+
+# Implementation of case option matching.
+#
+# This is the type of matching performed in a case option, using == for every type
+# of value except regular expression where a match is performed.
+#
+def is_match? left, right, o, scope
+  if right.is_a?(Regexp)
+    return false unless left.is_a? String
+    matched = right.match(left)
+    set_match_data(matched, scope) # creates or clears ephemeral
+    !!matched # convert to boolean
+  elsif right.is_a?(Puppet::Pops::Types::PAnyType)
```

Comment:
The above four lines can be replaced with a call to match_Regexp(right, left, scope)